api/src/main/java/org/openmrs/validator/DiagnosisValidator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+public class DiagnosisValidator implements Validator {
+
+	/**
+	 * @should support Diagnosis class
+	 */
+	@Override
+	public boolean supports(Class<?> aClass) {
+		return Diagnosis.class.isAssignableFrom(aClass);
+	}
+
+	@Override
+	public void validate(Object o, Errors errors) {
+		Diagnosis diagnosis = (Diagnosis)o;
+		
+		if(diagnosis == null){
+			throw new APIException("Diagnosis can't be null");
+		} else if (diagnosis.getVoided()) {
+			return;
+		}
+		
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "rank", "field.required");
```

Comment:
Are you in position to create some tests for this class? Take a look at these examples: https://github.com/openmrs/openmrs-core/tree/master/api/src/test/java/org/openmrs/validator

Reply:
I am looking at the examples and starting on the same.